pkgs/development/libraries/science/math/magma/generic.nix

```diff
@@ -119,15 +124,14 @@ stdenv.mkDerivation {
     cmake
     ninja
     gfortran
-  ];
+  ] ++ lists.optional cudaSupport cuda-native-redist;
```

Comment:
please prefer optionals. See https://discourse.nixos.org/t/lib-optional-considered-harmful/7081 for more info. The fact that optional exists at all is a bit unfortunate.